linfa-supervised/src/linear_regression/algorithm.rs

```diff
@@ -0,0 +1,119 @@
+#![allow(non_snake_case)]
+use ndarray::{stack, Array, Array1, ArrayBase, Axis, Data, Ix1, Ix2};
+use ndarray_linalg::Solve;
+/* I will probably change the implementation for an enum for more type safety.
+I have to make sure, it is a great idea when it comes to pyhton interoperability
```

Comment:
There is considerable freedom in how to wrap the Rust version for Python consumption - as detailed in #8, we shouldn't let Python move our Rust design in directions which are not idiomatic. The wrapping code can do the bridging when required 😀 

So I'd definitely suggest to go with the commented out version, which uses an enum (`LinearRegression::new(false)` is much more confusing than `LinearRegression::new(Intercept::NoIntercept)`).